src/python/espressomd/integrate.pyx

```diff
@@ -78,7 +80,14 @@ cdef class Integrator(object):
             check_type_or_throw_except(
                 reuse_forces, 1, bool, "reuse_forces has to be a bool")
 
-            _integrate(steps, recalc_forces, reuse_forces)
+            if (_integrate(steps, recalc_forces, reuse_forces)):
+                handle_errors("Encoutered errors during integrate")
+
+            if globals.set_py_interrupt:
+                PyErr_SetInterrupt()
+                globals.set_py_interrupt = False
+            PyErr_CheckSignals()
```

Comment:
`PyErr_CheckSignals()` could probably go into the `if` block as well.  Might save a few cycles.